backend/src/main/kotlin/blue/mild/covid/vaxx/service/VaccinationSlotService.kt

```diff
@@ -0,0 +1,127 @@
+    private val vaccinationSlotRepository: VaccinationSlotRepository,
+//    private val patientRepository: PatientRepository
+) {
+    private companion object : KLogging()
+
+    @Suppress("TooGenericExceptionThrown", "ThrowsCount")
+    suspend fun addSlots(createDto: CreateVaccinationSlotsDtoIn, locationId: EntityId? = null): List<EntityId> {
+        if (createDto.to < createDto.from.plusMillis(createDto.durationMillis)) {
+            throw Exception("Specified time range is not valid - ${createDto}")
+        }
+
+        if (createDto.locationId != null && locationId != null) {
+            throw Exception("Location mismatch - ${createDto} vs ${locationId}")
+        }
+        val usedLocationId = (createDto.locationId ?: locationId)!!
+
+        val location = locationRepository.getAndMapLocationsBy { Locations.id eq usedLocationId }
+            .singleOrNull() ?: throw entityNotFound<Locations>(Locations::id, usedLocationId)
+
+        var ts = createDto.from
+        val createdIds = mutableListOf<EntityId>()
+        while (ts.plusMillis(createDto.durationMillis).isBefore(createDto.to.plusMillis(1))) {
+            val to = ts.plusMillis(createDto.durationMillis)
+            for (queue in 0 until createDto.bandwidth) {
+                createdIds.add(
+                    vaccinationSlotRepository.addVaccinationSlot(
+                        locationId = location.id,
+                        patientId = null,
+                        queue = queue,
+                        from = ts,
+                        to = to,
+                    )
+                )
+            }
+            ts = to
+        }
+
+        return createdIds
+    }
+
+    /**
+     * Filters the database with the conjunction (and clause) of the given properties.
+     */
+    suspend fun getSlotsByConjunctionOf(
+        id: EntityId? = null,
+        locationId: EntityId? = null,
+        fromMillis: Long? = null,
+        toMillis: Long? = null,
+        status: VaccinationSlotStatus? = DEFAULT_STATUS,
+    ): List<VaccinationSlotDtoOut> {
+        // To Consider: Maybe when id is specified all the other parameter should be
+        val fromI = Instant.ofEpochMilli(fromMillis ?: Instant.EPOCH.toEpochMilli())
+        // MartinLLama: With Instant.MAX it was failing on Exception occurred in the application: long overflow
+        val toI = Instant.ofEpochMilli(toMillis ?: MAX_FUTURE.toEpochMilli())
+        val usedStatus = status ?: if (id == null) DEFAULT_STATUS else VaccinationSlotStatus.ALL
+
+        return vaccinationSlotRepository.get {
+            Op.TRUE
+                .andWithIfNotEmpty(id, VaccinationSlots.id)
+                .andWithIfNotEmpty(locationId, VaccinationSlots.locationId)
+                .and { VaccinationSlots.from.greaterEq(fromI) }
+                .and { VaccinationSlots.to.lessEq(toI) }
+                .and { when(usedStatus) {
+                    // MartinLLama: I do not know how to write ALL as match all = 1 == 1
```

Comment:
I'm not sure what do you mean by that?

Reply:
`when` has to be exhaustive. When the status is `ALL`, I do not want to do any additional filtering. In these situations I usually use something like `1 == 1`, but in this case I am not able to figure out how to write it, so I am checking column, that cannot be null whether it is not null => this is kind of wasteful.